pkg/reconciler/namespace/namespace.go

```diff
@@ -211,11 +225,15 @@ func (c *Controller) isValidCluster(ctx context.Context, lclusterName, clusterNa
 	if err != nil {
 		return false, err
 	}
-	isReady := conditions.IsTrue(cluster, clusterv1alpha1.ClusterReadyCondition)
-	if !isReady {
+	if ready := !conditions.IsTrue(cluster, clusterv1alpha1.ClusterReadyCondition); !ready {
```

Comment:
I think you mean `ready := conditions.IsTrue`?

Reply:
I caught this in the process of preparing #559.

Reply:
You may want to include https://github.com/kcp-dev/kcp/pull/559/commits/631c173a8f195759fe85201939ac95c93e62d6d9 in your PR:

 - ensures test updates to clusters are retried. Otherwise the test is prone to racing with cluster-updating controllers and updates will randomly fail. Other options including be patching or using SSA instead of `Update()` 
 - adds checks of cluster readiness before checking that a namespace is scheduled to a cluster. This ensures against timing issues with the controller responsible for setting cluster readiness.

Reply:
This is done now. If we're okay with this change as-is, we can just include your change in a followup PR. wdyt?

Reply:
My suggestion is to add that commit to this PR. Otherwise I think you'll have trouble passing e2e.

Reply:
I've incorporated your change (thanks for that!) and also changed Updates into Patches, to avoid races. PTAL.